src/hibernate-resume/hibernate-resume.c

```diff
@@ -29,17 +32,39 @@ int main(int argc, char *argv[]) {
                 return EXIT_SUCCESS;
 
         device = argv[1];
+        offset = getenv("SYSTEMD_HIBERNATE_RESUME_OFFSET");
 
         if (stat(device, &st) < 0) {
                 log_error_errno(errno, "Failed to stat '%s': %m", device);
                 return EXIT_FAILURE;
         }
 
         if (!S_ISBLK(st.st_mode)) {
-                log_error("Resume device '%s' is not a block device.", device);
+                log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Resume device '%s' is not a block device.", device);
                 return EXIT_FAILURE;
```

Comment:
hmm, if you add proper error codes, pleas rework the whole main() function to use DEFINE_MAIN_FUNCTION() like we do in all modern code these days.